### PR TITLE
fix(toolcall): use len(lower) not len(text) after ToLower to prevent out-of-bounds panic

### PR DESCRIPTION
**Pull request by waiwaic**
## Root Cause

`strings.ToLower` can **expand** byte length when processing certain Unicode characters. For example, the Turkish dotted I `İ` (U+0130, 2 bytes in UTF-8) lowercased to `i̇` (ASCII `i` + combining dot above U+0307) becomes **3 bytes**. After `lower := strings.ToLower(text)`, `len(lower)` can be larger than `len(text)`.

All loop bounds that iterate over `lower[n:]` slices must use `len(lower)`, never `len(text)`. Using `len(text)` when `len(lower) > len(text)` causes an out-of-bounds slice access and a panic.

## Changes

| File | Change |
|------|--------|
| `toolcalls_scan.go` | `ContainsToolCallWrapperSyntaxOutsideIgnored`: loop bound `len(text)` → `len(lower)` |
| `toolcalls_scan.go` | `FindToolMarkupTagOutsideIgnored`: loop bound `len(text)` → `len(lower)` |
| `toolcalls_scan.go` | `FindMatchingToolMarkupClose`: add `open.End >= len(text)` guard |
| `toolcalls_parse_markup.go` | `findXMLStartTagOutsideCDATA`: loop bound `len(text)` → `len(lower)` |
| `toolcalls_parse_markup.go` | `findMatchingXMLEndTagOutsideCDATA`: loop bound `len(text)` → `len(lower)` |
| `toolcalls_parse_markup.go` | `skipXMLIgnoredSection`: add `i < 0 \|\| i >= len(lower)` guard |
| `toolcalls_parse_markup.go` | `findToolCDATAEnd`: `from > len(text)` → `from >= len(text)` |

Also adds regression tests for Turkish I edge cases and `skipXMLIgnoredSection` boundary conditions.

Fixes #435.

### CODE REVIEW

File: internal/toolcall/toolcalls_scan.go [hunk 1]
```diff
@@ -57,7 +57,7 @@ func ContainsToolMarkupSyntaxOutsideIgnored(text string) (hasDSML, hasCanonical
 
 func ContainsToolCallWrapperSyntaxOutsideIgnored(text string) (hasDSML, hasCanonical bool) {
 	lower := strings.ToLower(text)
-	for i := 0; i < len(text); {
+	for i := 0; i < len(lower); {
```
[the remark this answers is not in the document]
Comment (waiwaic) [reply internal/toolcall/toolcalls_scan.go]: Agreed — min() works for the expand case but is still wrong for the shrink case. The better fix is to **remove the lower parameter from skipXMLIgnoredSection entirely** and let it construct `strings.ToLower(text[i:])` internally. This eliminates the boundary problem at the source. The outer tag-matching functions retain their own local `lower` for HasPrefix comparisons. See latest commit `ace4404`.

Also references the original issue #435.

File: internal/toolcall/toolcalls_parse_markup.go [hunk 2]
```diff
@@ -143,7 +143,7 @@ func findXMLElementBlocks(text, tag string) []xmlElementBlock {
 func findXMLStartTagOutsideCDATA(text, tag string, from int) (start, bodyStart int, attrs string, ok bool) {
 	lower := strings.ToLower(text)
 	target := "<" + strings.ToLower(tag)
-	for i := maxInt(from, 0); i < len(text); {
+	for i := maxInt(from, 0); i < len(lower); {
```
[the remark this answers is not in the document]
Comment (waiwaic) [reply internal/toolcall/toolcalls_parse_markup.go]: Same fix applies here — skipXMLIgnoredSection no longer takes a lower param. The len vs len(lower) boundary mismatch is gone by design. See commit `ace4404`.
